### PR TITLE
remove unused parameter in cpp basehandler

### PR DESCRIPTION
**Pull request by lxning**
## Description

Please read our [CONTRIBUTING.md](https://github.com/pytorch/serve/blob/master/CONTRIBUTING.md) prior to creating your first pull request.

Please include a summary of the feature or issue being fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

This PR is to 
- remove unused parameter msg in cpp basehandler
- update mnist handler example
- remove libmnist_handler.dylib from resource dir

Fixes #(issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Feature/Issue validation/testing

Please describe the Unit or Integration tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
Logs for Test A

- [ ] Test B
Logs for Test B

## Checklist:

- [ ] Did you have fun?
- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: cpp/src/examples/image_classifier/mnist/mnist_handler.cc [hunk 1]
```diff
@@ -7,9 +7,9 @@ void MnistHandler::Postprocess(
   for (const auto& kv : idx_to_req_id) {
     try {
       auto response = (*response_batch)[kv.second];
-      response->SetResponse(200, "data_tpye",
-                            torchserve::PayloadType::kDATA_TYPE_BYTES,
-                            torch::pickle_save(torch::argmax(data[kv.first])));
+      response->SetResponse(
+          200, "data_tpye", torchserve::PayloadType::kDATA_TYPE_BYTES,
+          torch::pickle_save(torch::argmax(data[kv.first].softmax(0))));
```
Comment (rohithkrn) [inline cpp/src/examples/image_classifier/mnist/mnist_handler.cc:12]: this would still return probabilities of all the classes instead of just the highest class, is that intended?

File: cpp/src/backends/torch_scripted/handler/base_handler.cc [hunk 2]
```diff
@@ -166,7 +166,6 @@ void BaseHandler::Postprocess(
   for (const auto& kv : idx_to_req_id) {
     try {
       auto response = (*response_batch)[kv.second];
-      auto msg = torch::argmax(data[kv.first]).to(torch::kFloat32);
       response->SetResponse(200, "data_type",
                             torchserve::PayloadType::kDATA_TYPE_BYTES,
                             torch::pickle_save(data[kv.first]));
```
Comment (rohithkrn) [inline cpp/src/backends/torch_scripted/handler/base_handler.cc:171]: why aren't we using softmax here?

Comment (lxning) [reply cpp/src/backends/torch_scripted/handler/base_handler.cc:171]: This is basehandler which can be used by any model, not image classify handler.

File: cpp/src/examples/image_classifier/mnist/mnist_handler.cc (hunk 1 above)
Comment (lxning) [reply cpp/src/examples/image_classifier/mnist/mnist_handler.cc]: this implementation is trying to match https://github.com/pytorch/serve/blob/master/examples/image_classifier/mnist/mnist_handler.py#L36

**Review by rohithkrn — APPROVED** [review]
